### ISSUE TITLE
When destroying a service-specific values.yaml environment override, the all-services values.yaml environment override is removed without warning

### ISSUE THREAD

**Issue by tylertechgraves** [harness/terraform-provider-harness#121]
### Terraform Version
1.1.4

### Harness provider version
LATEST

### Affected Resource(s)
Please list the resources as a list, for example:
- harness_yaml_config

### Terraform Configuration Files
```hcl
locals {
  name_prefix        = replace(lower(var.harness_application_name), " ", "-")
  git_connector_name = "${local.name_prefix}-${var.github_repository_name}"
}

data "harness_application" "harness_application" {
  name = var.harness_application_name
}

resource "harness_encrypted_text" "harness_sshsecret" {
  count = var.enabled ? 1 : 0
  name  = "${local.git_connector_name}-ssh-key"
  value = var.enabled ? (
    !var.ssh_key_in_secrets_manager ? var.private_key : null
  ) : null
  secret_manager_id = var.harness_secret_manager_id
  secret_reference = var.enabled ? (
    var.ssh_key_in_secrets_manager ? var.aws_secret_arn : null
  ) : null

  usage_scope {
    application_id          = data.harness_application.harness_application.id
    environment_filter_type = "NON_PRODUCTION_ENVIRONMENTS"
  }

  usage_scope {
    application_id          = data.harness_application.harness_application.id
    environment_filter_type = "PRODUCTION_ENVIRONMENTS"
  }
}

resource "harness_ssh_credential" "harness_sshcred" {
  count = var.enabled ? 1 : 0
  name  = local.git_connector_name
  ssh_authentication {
    port     = 22
    username = "git"
    inline_ssh {
      ssh_key_file_id = harness_encrypted_text.harness_sshsecret[0].id
    }
  }

  usage_scope {
    application_id          = data.harness_application.harness_application.id
    environment_filter_type = "NON_PRODUCTION_ENVIRONMENTS"
  }

  usage_scope {
    application_id          = data.harness_application.harness_application.id
    environment_filter_type = "PRODUCTION_ENVIRONMENTS"
  }

  # ssh_authentication isn't returned in the graphql query. This is a hack to keep this resource from being updated
  # on every apply.
  lifecycle {
    ignore_changes = [ssh_authentication]
  }
}

resource "harness_git_connector" "harness_gitconnector" {
  count          = var.enabled ? 1 : 0
  depends_on     = [github_repository_deploy_key.github_deploy_key]
  name           = local.git_connector_name
  url            = var.github_ssh_url
  url_type       = "REPO"
  branch         = var.github_default_branch
  username       = "git"
  ssh_setting_id = one(harness_ssh_credential.harness_sshcred[*].id)

  usage_scope {
    application_id          = data.harness_application.harness_application.id
    environment_filter_type = "NON_PRODUCTION_ENVIRONMENTS"
  }

  usage_scope {
    application_id          = data.harness_application.harness_application.id
    environment_filter_type = "PRODUCTION_ENVIRONMENTS"
  }
}

data "tls_public_key" "key" {
  count           = var.enabled && !var.ssh_key_in_secrets_manager ? 1 : 0
  private_key_pem = var.private_key
}

resource "github_repository_deploy_key" "github_deploy_key" {
  count      = var.enabled ? 1 : 0
  title      = "${local.git_connector_name} harness key (MANAGED BY TERRAFORM)"
  repository = var.github_repository_name
  key = var.enabled ? (
    var.ssh_key_in_secrets_manager ? var.ssh_public_key : data.tls_public_key.key[0].public_key_openssh
  ) : ""
  read_only = true
}

resource "harness_yaml_config" "service" {
  count   = var.enabled ? 1 : 0
  app_id  = var.use_application_id ? var.harness_application_id : one(data.harness_application.harness_application[*].id)
  path    = "Setup/Applications/${var.harness_application_name}/Services/${var.service_name}/Index.yaml"
  content = <<-EOF
    harnessApiVersion: '1.0'
    type: SERVICE
    artifactType: DOCKER
    deploymentType: KUBERNETES
    helmVersion: V2
    tags:
      svc: ${var.service_name}
    EOF
}

data "harness_git_connector" "connector" {
  count = length(var.git_connector_id) > 0 ? 1 : 0
  id    = var.git_connector_id
}

resource "harness_yaml_config" "service_manifest_remote" {
  count   = var.enabled ? 1 : 0
  app_id  = var.use_application_id ? var.harness_application_id : one(data.harness_application.harness_application[*].id)
  path    = "Setup/Applications/${var.harness_application_name}/Services/${var.service_name}/Manifests/Index.yaml"
  content = <<EOF
harnessApiVersion: '1.0'
type: APPLICATION_MANIFEST
gitFileConfig:
  branch: ${length(var.git_connector_branch) > 0 ? var.git_connector_branch : "master"}
  connectorName: %{if var.git_connector_id != ""}${data.harness_git_connector.connector[0].name}%{else}${var.git_connector_name}%{endif}
  %{~if var.git_connector_file_path != ""~}
  filePath: ${var.git_connector_file_path}
  %{~endif~}
  useBranch: true
  useInlineServiceDefinition: false
storeType: Remote
    EOF
}

resource "harness_yaml_config" "service_artifact_source" {
  count   = var.enabled ? 1 : 0
  app_id  = var.use_application_id ? var.harness_application_id : one(data.harness_application.harness_application[*].id)
  path    = "Setup/Applications/${var.harness_application_name}/Services/${var.service_name}/Artifact Servers/${var.service_name}.yaml"
  content = <<-EOF
    harnessApiVersion: '1.0'
    type: DOCKER
    imageName: ${var.container_artifact_source}
    serverName: ${var.harness_image_artifact_service_name}
    EOF
}

resource "harness_yaml_config" "harness_service_environment_overrides_remote" {
  count   = var.enabled && length(var.git_connector_name) > 0 ? 1 : 0
  app_id  = var.use_application_id ? var.harness_application_id : one(data.harness_application.harness_application[*].id)
  path    = "Setup/Applications/${var.harness_application_name}/Environments/${var.environment_id}/Values/Services/${var.service_name}/Index.yaml"
  content = <<EOF
harnessApiVersion: '1.0'
type: APPLICATION_MANIFEST_VALUES_ENV_SERVICE_OVERRIDE
gitFileConfig:
  branch: ${length(var.git_connector_branch) > 0 ? var.git_connector_branch : "master"}
  connectorName: ${var.git_connector_name}
  %{~if var.git_connector_file_path != ""~}
  filePath: ${var.git_connector_file_path}
  %{~endif~}
  useBranch: true
  useInlineServiceDefinition: false
storeType: Remote
    EOF
}
```

### Debug Output
Please note that in the above hcl, there's really only one resource related to this problem.  It's the last resource:
resource "harness_yaml_config" "harness_service_environment_overrides_remote" {}

We do not have debug output available from the destructive run.  We have attached the plan and apply logs, though (see References section below).

### Panic Output
Run does not panic.

### Expected Behavior
That only the specified service-specific values.yaml override would be removed as part of a destroy plan apply.

### Actual Behavior
The plan does NOT specify that the all-services values.yaml override on the environment will be destroyed, which is correct.  But when a destroy is run against the service-specific values.yaml harness environment override, the all-services values.yaml override on the environment is also destroyed unexpectedly.

### Steps to Reproduce
To reproduce:
1.) Run a destroy plan on a `harness_yaml_config` resource pointing at a service-specific values.yaml override.
2.) The plan produced is correct and shows the service-specific override being destroyed (along with the rest of the resources). The plan does NOT show that the all-services values.yaml override on the Harness environment will be removed.
3.) Apply the destroy plan.
4.) The service-specific values.yaml override IS removed correctly along with the rest of the resources in the template.
5.) The all-services values.yaml override in the Harness environment ALSO gets removed.

### Important Factoids
This is both a small and a huge problem for us.  It's a small problem, because we rarely destroy Harness resources.  It's a huge problem for us, because in the event that we destroy a service-specific values.yaml environment override, the all-services values.yaml environment override gets removed, and ALL our services depend on that (as the name implies).  This means that the destroy plan actually affects all the services that remain in harness after the destroy, even though we only wanted to destroy resources for a single harness service.

We suspect this could be an issue with the Harness API you guys call and may not be something you guys are doing wrong in this provider.  We already tried to report this to Harness, but they sent us here.

### References
I have attached the plan and apply logs from a run that caused the all-services values.yaml environment override to be incorrectly deleted.
[run-NeDaDn9b5Lhi49Fo-plan-log.txt](https://github.com/harness/terraform-provider-harness/files/8603991/run-NeDaDn9b5Lhi49Fo-plan-log.txt)
[run-NeDaDn9b5Lhi49Fo-apply-log.txt](https://github.com/harness/terraform-provider-harness/files/8603995/run-NeDaDn9b5Lhi49Fo-apply-log.txt)

**Comment by micahlmartin** [comment]
@tylertechgraves I'm having a hard time reproducing this issue. Here's what I setup:

- I have an existing application where I've configured an "All Services" override pointing at a remote values yaml file.
- In Terraform I create a service specific values override.

When I apply the Terraform I can see it successfully creates the service-specific override and after I delete it the "All Services" override is still in tact.

Here is my Terraform

```
data "harness_application" "this" {
  name = "Harness Automation Example"
}

resource "harness_yaml_config" "this" {
  app_id = data.harness_application.this.id
  path = "Setup/Applications/${data.harness_application.this.name}/Environments/dev/Values/Services/nginx/Index.yaml"
  content = <<EOF
harnessApiVersion: '1.0'
type: APPLICATION_MANIFEST_VALUES_ENV_SERVICE_OVERRIDE
gitFileConfig:
  branch: main
  connectorName: terraform-debug
  filePath: test.yaml
  useBranch: true
  useInlineServiceDefinition: false
storeType: Remote
  EOF
}
```

Can you try creating a very simplistic recreation using something similar to this and tell me if you get the same result? Or perhaps I'm just missing something in your description that you're doing differently.

**Comment by tylertechgraves** [comment]
@micahlmartin, thanks so much for looking at this.  I have not been able to reproduce using the latest version of this provider in conjunction with our latest terraform modules.  I can only reproduce using our older modules which were running on an earlier version of the provider.  I'm going to dig into this more and will let you know if I find something troubling.  The weirdest part for us was that the plan didn't show the removal was going to happen, and in Harness we saw the following:
![image](https://user-images.githubusercontent.com/16786974/167438504-0b141562-85e5-42fc-bb76-362238d7a77d.png)
The item at the top is the removal of our prod all-service override, and the item at the bottom is the removal of our qa all-services override.  Note that these are coming from 2600:1f18:218f:8e00:4ce:d328:64c7:1aa7 and 2600:1f16:c7f:3900:9d7c:5d85:d8a5:700a respectively.  These items appeared 2 minutes after applying terraform to remove a single service's environment override, and we still can't figure out where these calls came from.  We see no evidence it was something our terraform apply did (no indication it was going to be removed in the plan results or elsewhere).  But we did see this occur two times in the same day when running a destroy on two separate templates.
